LLama.Web/Models/LLamaModel.cs

```diff
@@ -1,108 +1,111 @@
-﻿using LLama.Abstractions;
+using LLama.Abstractions;
 using LLama.Web.Common;
 using System.Collections.Concurrent;
 
-namespace LLama.Web.Models
+namespace LLama.Web.Models;
+
+/// <summary>
+/// Wrapper class for LLamaSharp LLamaWeights
+/// </summary>
+/// <seealso cref="IDisposable" />
+public class LLamaModel : IDisposable
 {
+    private readonly ILogger _llamaLogger;
+    private readonly ModelOptions _config;
+    private readonly LLamaWeights _weights;
+    private readonly ConcurrentDictionary<string, LLamaContext> _contexts;
+
     /// <summary>
-    /// Wrapper class for LLamaSharp LLamaWeights
+    /// Initializes a new instance of the <see cref="LLamaModel"/> class.
     /// </summary>
-    /// <seealso cref="IDisposable" />
-    public class LLamaModel : IDisposable
+    /// <param name="modelParams">The model parameters.</param>
+    public LLamaModel(ModelOptions modelParams, ILogger llamaLogger)
     {
-        private readonly ILogger _llamaLogger;
-        private readonly ModelOptions _config;
-        private readonly LLamaWeights _weights;
-        private readonly ConcurrentDictionary<string, LLamaContext> _contexts;
-
-        /// <summary>
-        /// Initializes a new instance of the <see cref="LLamaModel"/> class.
-        /// </summary>
-        /// <param name="modelParams">The model parameters.</param>
-        public LLamaModel(ModelOptions modelParams, ILogger llamaLogger)
-        {
-            _config = modelParams;
-            _llamaLogger = llamaLogger;
-            _weights = LLamaWeights.LoadFromFile(modelParams);
-            _contexts = new ConcurrentDictionary<string, LLamaContext>();
-        }
-
-        /// <summary>
-        /// Gets the model configuration.
-        /// </summary>
-        public IModelParams ModelParams => _config;
-
-        /// <summary>
-        /// Gets the LLamaWeights
-        /// </summary>
-        public LLamaWeights LLamaWeights => _weights;
+        _config = modelParams;
+        _llamaLogger = llamaLogger;
+        _weights = LLamaWeights.LoadFromFile(modelParams);
```

Comment:
Async loading has been supported in #702, could you please introduce it here? Since async is not allowed in a class constructor, you could set the constructor as private and add a method `LLamaModel.CreateAsync` instead. For web app I think async loading is better because the UI would not be blocked when loading a large model.

Reply:
Will do.